src/Symfony/Component/DependencyInjection/Alias.php

```diff
@@ -85,40 +83,71 @@ public function isPrivate()
      * Whether this alias is deprecated, that means it should not be referenced
      * anymore.
      *
-     * @param bool   $status   Whether this alias is deprecated, defaults to true
-     * @param string $template Optional template message to use if the alias is deprecated
+     * @param string $package The name of the composer package that is triggering the deprecation
+     * @param string $version The version of the package that introduced the deprecation
+     * @param string $message The deprecation message to use
      *
      * @return $this
      *
      * @throws InvalidArgumentException when the message template is invalid
      */
-    public function setDeprecated(bool $status = true, string $template = null)
+    public function setDeprecated(/* string $package, string $version, string $message */)
     {
-        if (null !== $template) {
-            if (preg_match('#[\r\n]|\*/#', $template)) {
+        $args = \func_get_args();
+
+        if (\func_num_args() < 3) {
+            trigger_deprecation('symfony/dependency-injection', '5.1', 'The signature of method "%s()" requires 3 arguments: "string $package, string $version, string $message", not defining them is deprecated.', __METHOD__);
```

Comment:
we need a separate message when $status = false, as spotted by @stof in #35871

Reply:
done